keyvault/client/secrets/src/main/java/com/azure/keyvault/SecretClientBuilder.java

```diff
@@ -0,0 +1,215 @@
+     *
+     * @param credentials The credentials to use for authenticating HTTP requests.
+     * @return the updated Builder object.
+     * @throws NullPointerException if {@code credentials} is {@code null}.
+     */
+    public SecretClientBuilder credentials(ServiceClientCredentials credentials) {
+        Objects.requireNonNull(credentials);
+        this.credentials = credentials;
+        return this;
+    }
+
+    /**
+     * Sets the logging level for HTTP requests and responses.
+     *
+     * <p>logLevel is optional. If not provided, default value of {@link HttpLogDetailLevel#NONE} is set.</p>
+     *
+     * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+     * @return the updated Builder object.
+     * @throws NullPointerException if {@code logLevel} is {@code null}.
+     */
+    public SecretClientBuilder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+        Objects.requireNonNull(logLevel);
+        httpLogDetailLevel = logLevel;
+        return this;
+    }
+
+    /**
+     * Adds a policy to the set of existing policies that are executed after
+     * {@link SecretClient} required policies.
+     *
+     * @param policy The {@link HttpPipelinePolicy policy} to be added.
+     * @return the updated Builder object.
+     * @throws NullPointerException if {@code policy} is {@code null}.
+     */
+    public SecretClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy);
+        policies.add(policy);
+        return this;
+    }
+
+    /**
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
+     *
+     * @param client The HTTP client to use for requests.
+     * @return the updated Builder object.
+     * @throws NullPointerException If {@code client} is {@code null}.
+     */
+    public SecretClientBuilder httpClient(HttpClient client) {
+        Objects.requireNonNull(client);
+        this.httpClient = client;
+        return this;
+    }
+
+    /**
+     * Sets the HTTP pipeline to use for the service client.
+     *
+     * If {@code pipeline} is set, all other settings are ignored, aside from
+     * {@link SecretClientBuilder#endpoint(String) endpoint} to build {@link SecretClient}.
+     *
+     * @param pipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return the updated {@link SecretClientBuilder} object.
+     */
+    public SecretClientBuilder pipeline(HttpPipeline pipeline) {
+        Objects.requireNonNull(pipeline);
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    private TokenCredentials getTokenCredentials() {
+        String token = "";
+        try {
+            token = credentials.authorizationHeaderValue(endpoint.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return new TokenCredentials("Bearer", token);
+    }
```

Comment:
This is ok for the initial check-in but it will have to change once we have the updated authentication classes.  As is this will not allow for the refreshing of the token across the lifetime of the client, once the token expires all calls will fail.